src/main/java/com/olegpruh/bookstore/repository/book/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,21 @@
+package com.olegpruh.bookstore.repository.book;
+
+import com.olegpruh.bookstore.model.Book;
+import com.olegpruh.bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) -> root.get("author")
+                .in(Arrays.stream(params).toArray());

```

Comment:
Why do we need to convert it to stream and back?
```suggestion
                .in(params);
```

Reply:
@sokimaaa The IDE says: “Confusing argument 'params', unclear if a varargs or non-varargs call is desired”. Therefore, I used Stream as well as in the video.